python/hsfs/feature_view.py

```diff
@@ -0,0 +1,580 @@
+        statistics_config: Optional[Union[StatisticsConfig, bool, dict]] = None,
+        read_options: Optional[Dict[Any, Any]] = {}
+    ):
+        """
+        # Returns
+            Training datasets: `dict(str, Dataframe)`
+                Dictionary of dataframes where split is the key
+
+        """
+        td = training_dataset.TrainingDataset(
+            name=self.name,
+            version=version,
+            event_start_time=start_time,
+            event_end_time=end_time,
+            description=description,
+            storage_connector=None,
+            featurestore_id=self._featurestore_id,
+            data_format="tsv",
+            location="",
+            splits=splits,
+            statistics_config=statistics_config
+        )
+        # td_job is used only if the python engine is used
+        td, df = self._feature_view_engine.get_training_data(
+            self, td, read_options, splits=splits
+        )
+        if version is None:
+            warnings.warn(
+                "No version provided for creating training dataset, incremented version to `{}`.".format(
+                    td.version
+                ),
+                util.VersionWarning,
+            )
+        return td.version, df
+
+    def get_training_dataset(
+        self,
+        start_time: Optional = None,
+        end_time: Optional = None,
+        version: Optional[int] = None,
+        description: Optional[str] = "",
+        statistics_config: Optional[Union[StatisticsConfig, bool, dict]] = None,
+        read_options: Optional[Dict[Any, Any]] = {}
+    ):
+        """ Get training data from storage or feature groups.
+
+        If version is not provided or provided version has not already existed, it creates
+        a new version of training data according to given arguments and returns a dataframe.
+
+        If version is provided and has already existed, it reads training data from storage
+        or feature groups and returns a dataframe. If split is provided, it read the specific split.
+
+        !!! info
+        If a materialised training data has deleted. Use `recreate_training_dataset()` to
+        recreate the training data.
+
+        start_time: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
+                following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, or `%Y%m%d%H%M%S`.
+        end_time: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
+                following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, or `%Y%m%d%H%M%S`.
+        """
+        td, df = self.get_training_datasets(
+            {},
+            start_time=start_time,
+            end_time=end_time,
+            version=version,
+            description=description,
+            statistics_config=statistics_config,
+            read_options=read_options,
+        )
+        return td, df
+
+    def create_training_dataset(
```

Comment:
Why did you skip transformation functions here?

Reply:
because transformation is defined by feature view not training dataset.

Reply:
alright, makes sense.  then remove them from the api doc